pkg/drift/drift.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drift
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/abcxyz/guardian/pkg/drift/assets"
+	"github.com/abcxyz/guardian/pkg/drift/iam"
+	"github.com/abcxyz/guardian/pkg/drift/terraform"
+)
+
+// Process compares the actual GCP IAM against the IAM in your Terraform state files.
+func Process(ctx context.Context, organizationID, bucketQuery string) error {
+	assetsClient, err := assets.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize assets client: %w", err)
+	}
+	folders, err := assetsClient.HierarchyAssets(ctx, organizationID, assets.FolderAssetType)
+	if err != nil {
+		return fmt.Errorf("failed to get folders: %w", err)
+	}
+	projects, err := assetsClient.HierarchyAssets(ctx, organizationID, assets.ProjectAssetType)
+	if err != nil {
+		return fmt.Errorf("failed to get folders: %w", err)
+	}
```

Comment:
Why does this need to be two different calls versus asking for all folders and projects in a single query?

Reply:
Okay, can we comment that inline?